### ISSUE TITLE
ofxTCPServer joining data into single trasnmition on Raspberry Pi 4

### ISSUE THREAD

**Issue by leonyuhanov** [openframeworks/openFrameworks#7603]
I have an OF app that is sending X number of bytes to an ESP32 client at 25fps.
If the framerate is above 5fps the server sometimes joins data sends in to a single transmition. So instead od receiving X number of bytes. I get a multiple like 2x or higher in faster transmition rates.
This occurs because TCP has TCP_NODELAY Option enabled by default. This may not be an "issue" for everyone but was definitely a problem for my aplication.
I resolved this by adding the following lines of code into the Create() function in ofxTCPManager.cpp when the tcp conwction is beeing made:

```
bool ofxTCPManager::Create()
{
	if (m_hSocket != INVALID_SOCKET) return(false);
	m_closing = false;
	m_hSocket = socket( AF_INET, SOCK_STREAM, IPPROTO_IP);
	**int enable=1;
	if (setsockopt(m_hSocket,IPPROTO_TCP,TCP_NODELAY,(char*)&enable,sizeof(int))< 0)	
	{
		//adedd to disable no delay
	}**	
	bool ret = (m_hSocket != INVALID_SOCKET);
	if(!ret) ofxNetworkCheckError();
	return ret;
}
```

**Comment by dimitre** [comment]
I agree this should be easily set without editing the core. maybe a new ofxTCPManagerSettings object?

**Comment by artificiel** [comment]
@leonyuhanov are you using ofxTCPClient/Server or are you directly interfacing with your own instances of ofxTCPManager?

(I think in most use cases ofxTCPManager is hidden from the user interface, and it would be tricky to propagate a settings into it.)

the option could be added to ofxTCPSettings (which can be declared adjacent to the ofxTCP{Client|Server}::setup() call site) and passed as a bool argument to `ofxTCPManager::Create()`?

**Comment by leonyuhanov** [comment]
@artificiel I was using ofxTCPServer
Yes I thing a new settings option would be useful! ,although i think this is an edge case, not sure how many people have ever been effected by this :)

**Comment by artificiel** [comment]
in general the typical use for small low-latency grains kind of traffic is UDP (like OSC), but TCP_NODELAY makes TCP almost as responsive as UDP — [it's used for instance in ofxMQTT](https://github.com/search?q=repo%3A256dpi%2FofxMQTT%20TCP_NODELAY&type=code) (which makes use of mosquito's network stack; not ofxTCP) 

so it's a good option to add, and might allow some low-latency ideas to be implemented over TCP. adding an option in ofxTCPSettings it non-obtrusive and won't affect previous behaviour.

I can't edit your post title; can you rename it to something like "add option in ofxTCPSettings to allow TCP_NODELAY" so it looks more like a "todo" item? and we'll keep the discussion above as reference.

**Comment by leonyuhanov** [comment]
@artificiel done!
I ended up swapping my whole project over to UDP anyway, but hey if it helps anyone else!
